core/src/main/java/hudson/Functions.java

```diff
@@ -1903,6 +1904,16 @@ public static String joinPath(String... components) {
             return joinPath(Stapler.getCurrentRequest().getContextPath() + '/' + itUrl, urlName);
     }
 
+    /**
+     * Computes the link to the console for the specified build, taking {@link ConsoleUrlProvider} into account.
+     * @param run the build
+     * @return the absolute URL for accessing the specified build's console
+     * @since TODO
+     */
+    public static @CheckForNull String getConsoleUrl(Run<?, ?> run) {
```

Comment:
This method is intended for use in Jelly files (both in core and in plugins). I think it is a little cleaner to put it here rather than directly on `Run`, but that is just my opinion.